### PR TITLE
🚀 refactor: improve logic method BodyParser

### PR DESCRIPTION
**Pull request by renanbastos93**
* Changed many `if` to a `switch case`
* Removed unused variable
* Defined alias tag when needed

### CODE REVIEW

**Comment by jozsefsallai** [comment]
This looks fine **but** I'll have to check the behavior of this when there are other tokens appended to the content type (iirc, multipart/form-data does that a lot when appending the boundary).

File: ctx.go [hunk 1]
```diff
@@ -222,47 +222,44 @@ func (ctx *Ctx) Body() string {
...
-	// application/json
-	if strings.HasPrefix(ctype, MIMEApplicationJSON) {
+	var (
+		schemaDecoder = schema.NewDecoder()
+		ctype         = getString(ctx.Fasthttp.Request.Header.ContentType())
+	)
+	schemaDecoder.IgnoreUnknownKeys(true)
+
+	switch ctype {
+	case MIMEApplicationJSON: // application/json
```
Comment (Fenny) [inline ctx.go:232]: Please add the `utf-8` variant
```go
case MIMEApplicationJSON, MIMEApplicationJSONUTF8:
```

File: ctx.go [hunk 2]
```diff
@@ -222,47 +222,44 @@ func (ctx *Ctx) Body() string {
...
 		return json.Unmarshal(ctx.Fasthttp.Request.Body(), out)
-	}
-	// application/xml text/xml
-	if strings.HasPrefix(ctype, MIMEApplicationXML) || strings.HasPrefix(ctype, MIMETextXML) {
+	case MIMEApplicationXML: // application/xml
```
Comment (Fenny) [inline ctx.go:234]: Please add the `utf-8` variant
```go
case MIMEApplicationXML, MIMEApplicationXMLUTF8:
```

File: ctx.go [hunk 3]
```diff
@@ -222,47 +222,44 @@ func (ctx *Ctx) Body() string {
...
 		return xml.Unmarshal(ctx.Fasthttp.Request.Body(), out)
-	}
-	// application/x-www-form-urlencoded
-	if strings.HasPrefix(ctype, MIMEApplicationForm) {
+	case MIMETextXML: // text/xml
```
Comment (Fenny) [inline ctx.go:236]: Please add the `utf-8` variant
```go
case MIMETextXML, MIMETextXMLUTF8:
```

**Comment by Fenny** [comment]
> multipart/form-data

This is correct, the `multipart/form-data` header could be unique on every request i.e. 
`Content-Type: multipart/form-data; boundary=something`

We should move this one outside the switch statement and perform a `Contains` check.

**Comment by renanbastos93** [comment]
Alright I will improve it, thanks for feedbacks

**Comment by renanbastos93** [comment]
> This looks fine **but** I'll have to check the behavior of this when there are other tokens appended to the content type (iirc, multipart/form-data does that a lot when appending the boundary).
> 
> The original code checks if the supplied content type starts with the XML, JSON, etc. content types, not that they're equal.

Nice catch, thanks I improved. Please review guys

**Comment by Fenny** [comment]
Awesome work, thnx!

**Pull request merged**